File: src/main/java/com/paragon/client/managers/alt/Alt.java
package com.paragon.client.managers.alt;

import com.paragon.asm.mixins.accessor.IMinecraft;
import com.paragon.client.ui.alt.AltManagerGUI;
import fr.litarvan.openauth.microsoft.MicrosoftAuthResult;
import fr.litarvan.openauth.microsoft.MicrosoftAuthenticationException;
import fr.litarvan.openauth.microsoft.MicrosoftAuthenticator;
import net.minecraft.client.Minecraft;
import net.minecraft.util.Session;
import net.minecraft.util.text.TextFormatting;

public class Alt {

    private String email;
    private String password;
    private Session session;

    public Alt(String email, String password) {
        this.email = email;
        this.password = password;
    }

    /**
     * Gets the email of the alt
     *
     * @return The email of the alt
     */
    public String getEmail() {
        return email;
    }

    /**
     * Gets the password of the alt
     *
     * @return The password of the alt
     */
    public String getPassword() {
        return password;
    }

    /**
     * Logs the user into an alt account
     */
    public boolean login() {
        if (session == null) {
            // Authenticator
            MicrosoftAuthenticator auth = new MicrosoftAuthenticator();
            System.out.println("logging in as " + this.email);
            try {
                // Get auth result
                MicrosoftAuthResult result = auth.loginWithCredentials(this.email, this.password);
                // Set alt session
                this.session = new Session(result.getProfile().getName(), result.getProfile().getId(), result.getAccessToken(), "legacy");
            } catch (MicrosoftAuthenticationException microsoftAuthenticationException) {
                microsoftAuthenticationException.printStackTrace();
            }
        }

        // Return false if the session is null
        if (this.session == null) {
            AltManagerGUI.renderString = TextFormatting.RED + "Unsuccessful Login!";
            return false;
        }

        // Set Minecraft session
        ((IMinecraft) Minecraft.getMinecraft()).setSession(this.session);
        AltManagerGUI.renderString = TextFormatting.GREEN + "Successful Login!";

        return true;
    }
}

File: src/main/kotlin/com/paragon/client/managers/alt/Alt.kt
package com.paragon.client.managers.alt

import com.paragon.Paragon
import com.paragon.mixins.accessor.IMinecraft
import com.paragon.client.ui.alt.AltManagerGUI
import fr.litarvan.openauth.microsoft.MicrosoftAuthenticationException
import fr.litarvan.openauth.microsoft.MicrosoftAuthenticator
import net.minecraft.client.Minecraft
import net.minecraft.util.Session
import net.minecraft.util.text.TextFormatting

/**
 * @author Surge, SooStrator1136
 */
class Alt(val email: String, val password: String) {

    private var session: Session? = null

    /**
     * Logs the user into an alt account
     */
    fun login(): Boolean {
        if (session == null) {
            val auth = MicrosoftAuthenticator() //Authenticator
            Paragon.INSTANCE.logger.info("logging in with $email")
            try {
                val result = auth.loginWithCredentials(email, password) //Get auth result
                session = Session(result.profile.name, result.profile.id, result.accessToken, "legacy") //Set alt session
            } catch (e: MicrosoftAuthenticationException) {
                e.printStackTrace()
            }
        }

        //Return false if the session is null
        if (session == null) {
            AltManagerGUI.renderString = TextFormatting.RED.toString() + "Unsuccessful Login!"
            return false
        }

        //Set Minecraft session
        (Minecraft.getMinecraft() as IMinecraft).session = session
        AltManagerGUI.renderString = TextFormatting.GREEN.toString() + "Successful Login!"

        return true
    }

}